Logging/.CSProject/Logger/OneTimeLogger.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Anvil.CSharp.Logging
+{
+    /// <summary>
+    /// A wrapper for <see cref="Logger"/> that only allows a message to be emitted from a given call site once per session.
+    /// A call site is determined to be unique by a combination of its file path and line number.
+    /// </summary>
+    public readonly struct OneTimeLogger : ILogger
+    {
+        private static readonly HashSet<int> s_CalledLogSites = new HashSet<int>();
+
+        /// <summary>
+        /// Resets <see cref="OneTimeLogger"/>'s to allow one time logs to be emitted one more time.
+        /// </summary>
+        public static void Reset()
+        {
+            s_CalledLogSites.Clear();
+        }
```

Comment:
Is there a use case for this? While I'm not sure how it could be done, allowing a single log to appear again seems useful, resetting all logs across the project less so

Reply:
Alright, I've read the Unity PR now :P I should've guessed that this covers an editor-only problem, and isn't likely to be actually used in a normal context.

Reply:
😄 Yea. Unity domain reload specific feature.

I could see it potentially used in a situation where an application soft resets (post update?) but even then, depending on how the application domains are managed the newly loaded code would likely have its own version of logger.